packages/block-library/src/columns/column.js

```diff
@@ -1,12 +1,66 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
 import { Path, SVG } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { InnerBlocks } from '@wordpress/block-editor';
+import { InnerBlocks, BlockControls, BlockVerticalAlignmentToolbar } from '@wordpress/block-editor';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
 
 export const name = 'core/column';
 
+const ColumnEdit = ( { attributes, updateAlignment } ) => {
+	const { verticalAlignment } = attributes;
+
+	const classes = classnames( 'block-core-columns', {
+		[ `is-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
+	} );
+
+	const onChange = ( alignment ) => updateAlignment( alignment );
+
+	return (
+		<div className={ classes }>
+			<BlockControls>
+				<BlockVerticalAlignmentToolbar
+					onChange={ onChange }
+					value={ verticalAlignment }
+				/>
+			</BlockControls>
+			<InnerBlocks templateLock={ false } />
+		</div>
+	);
+};
+
+const edit = compose(
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockRootClientId } = select( 'core/editor' );
+
+		return {
+			parentColumsBlockClientId: getBlockRootClientId( clientId ),
```

Comment:
Noting:

- Typo on "Colums" ("Columns")
- `getBlockRootClientId` is a very non-performant selector in its current form. Ideally this shouldn't impact you in implementing this, but unfortunately it might. Options here may be to either improve the performance of (at worst, memoize) the selector, or pass through the _selector_, not the selector result, to be called directly in the callback of `updateAlignment` when it's needed.

Reply:
> * `getBlockRootClientId` is a very non-performant selector in its current form. Ideally this shouldn't impact you in implementing this, but unfortunately it might. Options here may be to either improve the performance of (at worst, memoize) the selector, or pass through the _selector_, not the selector result, to be called directly in the callback of `updateAlignment` when it's needed.

Revisiting this since I'm encountering it again: Since it's only used in the `withDispatch` callback, another (preferred) option would be to leverage the _third_ argument of `withDispatch` (`registry`) to select the necessary data on-demand only when necessary. I suspect this would provide a significant performance improvement for posts containing a columns block.

See: #11851

For what it's worth, I'm refactoring this in an unrelated effort, so it may not require any action on your part.